app/main/activity/crowdsourced-survey-table.directive.js

```diff
@@ -0,0 +1,97 @@
+module.exports = CrowdsourcedSurveyTable;
+
+function CrowdsourcedSurveyTable() {
+    return {
+        restrict: 'E',
+        scope: {
+            filters: '='
+        },
+        controller: CrowdsourcedSurveyTableController,
+        template: require('./crowdsourced-survey-table.html')
+    };
+}
+
+CrowdsourcedSurveyTableController.$inject = ['$scope', '$translate', 'FormEndpoint', 'FormStatsEndpoint', 'FormContactEndpoint', 'PostEndpoint', '_', 'PostFilters', 'Session'];
+function CrowdsourcedSurveyTableController($scope, $translate, FormEndpoint, FormStatsEndpoint, FormContactEndpoint, PostEndpoint, _, PostFilters, Session) {
+    $scope.crowdsourcedSurveyStatsByForm = [];
+    $scope.totalAllReduced = 0;
+    $scope.totalWebReduced = 0;
+    $scope.totalEmailReduced = 0;
+    $scope.totalSmsReduced = 0;
+    $scope.totalTwitterReduced = 0;
+    $scope.noSurveys = false;
+    $scope.unstructuredStats = {};
+
+    // whenever the filters changes, update the current list of posts
+    // this also initiates the first call to get form stats because
+    // it is called when the filters load the first time
+    $scope.$watch('filters', function () {
+        clearStats();
+        getFormStats();
+    }, true);
+    PostFilters.setMode('activity');
+
+    function clearStats() {
+        $scope.crowdsourcedSurveyStatsByForm = [];
+        $scope.totalAllReduced = 0;
+        $scope.totalWebReduced = 0;
+        $scope.totalEmailReduced = 0;
+        $scope.totalSmsReduced = 0;
+        $scope.totalTwitterReduced = 0;
+        $scope.unstructuredStats = {};
+    }
+
+    function getFormStats() {
+        FormEndpoint.query().$promise.then((forms) => {
+            if (crowdsourcedSurveysExist) {
+                $scope.noSurveys = false;
+                forms.forEach((form) => {
+                    if (!form.targeted_survey) {
+                        let query = PostFilters.getQueryParams($scope.filters) || {};
+                        var postQuery = _.extend({}, query, {
+                            formId: form.id
+                        });
+
+                        FormStatsEndpoint.query(postQuery).$promise.then((stats) => {
+                            stats.name = form.name;
```

Comment:
I think you could do `                            $scope.crowdsourcedSurveyStatsByForm.push(form.name);` directly

Reply:
I need to add the name to the object that I'm pushing to the array, rather than just the name itself.